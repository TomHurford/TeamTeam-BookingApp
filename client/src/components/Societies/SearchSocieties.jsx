import React, { Component } from "react";
import Pagination from "../common/Pagination";

import { paginate } from "../../utils/paginate";
import SearchBar from "../common/Searchbar";
import { Link } from "react-router-dom";

class SearchSocieties extends Component {
  constructor(props) {
    super(props);
    this.state = {
      data: [], // bad implementation, will change later
      currentPage: 1,
      pageSize: 5,
      searchQuery: " ",
    };
    this.fetchData();
  }

  fetchData() {
    fetch("http://localhost:5001/societies/getSocieties")
      .then((response) => response.json())
      .then((societiesList) => {
        this.setState({ data: societiesList });
      })
      .catch((error) => console.error(error));
  }

  handlePageChange = (page) => {
    this.setState({ currentPage: page });
  };

  handleSearch = (query) => {
    this.setState({ searchQuery: query, currentPage: 1 });
  };

  render() {
    const { pageSize, currentPage } = this.state;

    const societies = paginate(this.state.data, currentPage, pageSize);

    return (
      <React.Fragment>
        <h2>Societies</h2>
        <SearchBar onChange={this.handleSearch} />

        <table className="table">
          <thead>
            <tr>
              <th>Name</th>
              <th>Category</th>
              <th>Followers</th>
              <th>Number of Events</th>
            </tr>
          </thead>
          <tbody>
            {societies.map((society) => (
              <tr key={society.id}>
                <td>
                  <Link to={`/society/${society.id}`}>{society.name}</Link>
                </td>
                <td>{society.category}</td>
                <td>{society.members}</td>
                <td>{society.description}</td>
              </tr>
            ))}
          </tbody>
        </table>
        <Pagination
          itemsCount={this.state.data.length}
          pageSize={this.state.pageSize}
          currentPage={this.state.currentPage}
          onPageChange={this.handlePageChange}
        />
        <Link to="/create-society">
          <button className="btn btn-primary" style={{ marginRight: "15px" }}>
            Create Society
          </button>
        </Link>
        <Link to="/edit-society">
          <button className="btn btn-primary">Edit Society</button>
        </Link>
        {/* TODO: Use destructuring */}
      </React.Fragment>
    );
  }
}

export default SearchSocieties;
