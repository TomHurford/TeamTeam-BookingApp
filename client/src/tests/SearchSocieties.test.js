import React from "react";
import renderer from "react-test-renderer";
import SearchSocieties from "../components/Societies/SearchSocieties";
import { BrowserRouter } from "react-router-dom";
// Snapshot test for searching societies
test("SearchSocieties snapshot test", () => {
  const component = renderer.create(
    <BrowserRouter>
      <SearchSocieties />
    </BrowserRouter>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});