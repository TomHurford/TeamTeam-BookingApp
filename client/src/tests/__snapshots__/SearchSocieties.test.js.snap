// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchSocieties snapshot test 1`] = `
<div
  style={
    Object {
      "marginLeft": "8px",
      "marginTop": "60px",
    }
  }
>
  <h1>
    Societies
  </h1>
  <div
    data-testid="searchbar"
  >
    <input
      className="form-control my-4"
      name="query"
      onChange={[Function]}
      placeholder="Search for society..."
      type="text"
      value=""
    />
  </div>
  <div
    className="row"
  >
    <div
      className="col-2"
    >
      <ul
        className="list-group"
      >
        <li
          className="list-group-item"
          onClick={[Function]}
        >
          All
        </li>
        <li
          className="list-group-item"
          onClick={[Function]}
        >
          Sports
        </li>
        <li
          className="list-group-item"
          onClick={[Function]}
        >
          Academic
        </li>
        <li
          className="list-group-item"
          onClick={[Function]}
        >
          Social
        </li>
        <li
          className="list-group-item"
          onClick={[Function]}
        >
          Other
        </li>
      </ul>
    </div>
    <div
      className="col"
    >
      <table
        className="table"
      >
        <thead>
          <tr>
            <th>
              Name
            </th>
            <th>
              Category
            </th>
            <th>
              Followers
            </th>
            <th>
              Description
            </th>
          </tr>
        </thead>
        <tbody />
      </table>
      <div
        data-testid="pagination"
      >
        <nav>
          <ul
            className="pagination"
          />
        </nav>
      </div>
      <a
        href="/create-society"
        onClick={[Function]}
      >
        <button
          className="btn btn-primary"
          style={
            Object {
              "marginRight": "15px",
            }
          }
        >
          Create Society
        </button>
      </a>
      <a
        href="/edit-society"
        onClick={[Function]}
      >
        <button
          className="btn btn-primary"
        >
          Edit Society
        </button>
      </a>
    </div>
  </div>
</div>
`;
